IOMC/RandomEngine/src/XorShift128PlusAdaptor.cc

```diff
@@ -0,0 +1,213 @@
+#include "IOMC/RandomEngine/src/XorShift128PlusAdaptor.h"
+#include "FWCore/Utilities/interface/EDMException.h"
+
+#include "CLHEP/Random/engineIDulong.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "TBufferFile.h"
+
+#include <cstddef>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <stdint.h>
+#include <string>
+#include <time.h>
+
+
+namespace edm {
+
+XorShift128PlusAdaptor::XorShift128PlusAdaptor() {
+  
+  srand(time(NULL));
+  for( auto& seed : seeds ) {
+    seed = rand();
+  }
+}
+
+XorShift128PlusAdaptor::XorShift128PlusAdaptor( const std::array<uint64_t,2>& inseeds ) {  
+  if( inseeds.size() < 2 ) {
+    Grumble(std::string("XorShift128Plus needs two 64 bit seeds!"));
+  }
+  
+  if( inseeds[0] == 0 && inseeds[1] == 0 ) {
+    Grumble(std::string("XorShift128Plus cannot be seeded with all zeroes!"));
+  }
+  seeds = inseeds;
+  state = seeds;
+  theSeeds = reinterpret_cast<const int64_t*>(seeds.data());
+}
+
+XorShift128PlusAdaptor::XorShift128PlusAdaptor( const std::array<uint32_t,4>& inseeds ) {  
+  if( inseeds.size() < 4 ) {
+    Grumble(std::string("XorShift128Plus needs four 32 bit seeds!"));
+  }
+  
+  if( inseeds[0] == 0 && inseeds[1] == 0 && 
+      inseeds[2] == 0 && inseeds[2] == 0    ) {
+    Grumble(std::string("XorShift128Plus cannot be seeded with all zeroes!"));
+  }
+
+  seeds.fill(0ULL);
+  for( size_t i = 0; i < inseeds.size() && i < 4; ++i ) {
+    uint64_t temp = inseeds[i];
+    seeds[i/2] = ( temp <<( i%2 ? 0 : 32 ) ); 
+  }
+  state = seeds;
+  theSeeds = reinterpret_cast<const int64_t*>(seeds.data());
+}
+
+XorShift128PlusAdaptor::XorShift128PlusAdaptor(std::istream&) {
+  Grumble(std::string("Cannot instantiate a TRandom engine from an istream"));
+}
```

Comment:
Should change the references from TRandom to XorShift128Plus.  Just wondering, did you try TRandom3 instead of HepJames and was it also too slow? I do not know the answer, I was just wondering.